Dockerfile

```diff
@@ -0,0 +1,63 @@
+# syntax=docker/dockerfile:1
+
+ARG GO_VERSION=1.23.7
+ARG LLAMA_SERVER_VERSION=latest
+ARG LLAMA_BINARY_PATH=/com.docker.llama-server.native.linux.cpu.amd64
+
+FROM golang:${GO_VERSION}-alpine AS builder
```

Comment:
I'd also aim to use a glibc-compatbile Go image for the compilation too (even with `CGO_ENABLED=false`).  If we use `ubuntu:24.04` for the production images, then `golang:1.24.2-bookworm` tag should be fine (Debian Bookworm ships with glibc 2.36 / GCC 12.2.0, Ubuntu 24.04 ships with glibc 2.39 / GCC 13.2).